### DISCUSSION TITLE
Ruff for linting and formatting

### DISCUSSION THREAD

**Discussion by bluginbuhl** [drivendataorg/cookiecutter-data-science#374] [Requested tools]
Currently, the generated cookie-cutter project uses `black`, `ruff.isort`, and `flake8`. These tools are configured in different locations (`pyproject.toml` and `setup.cfg`). These configurations could all be consolidated to use `ruff`. Additionally the base repo could also use `ruff` for formatting and linting, which would simplify the dev-requirements.

**Comment by chrisjkuch** [comment]
Two levels of decisions here:
* Do we add support for ruff? (I think yes)
* Do we make ruff the default for everything (formatting + linting)? (I also think yes)

In this world, there would be two or three options:
1. ruff for everything (default)
2. black + flake8 + isort
3. nothing?

**Comment by chrisjkuch** [comment]
Support for ruff was added #387 - it's now the default linting & formatting option. It will be a part of the 2.1.0 release.

If you don't want to wait, you can use the most recent version of the template by pulling from the master branch, `ccds -c master`

**Comment by chrisjkuch** [comment]
[v2.1.0 released](https://github.com/drivendataorg/cookiecutter-data-science/releases/tag/v2.1.0)!
